R/annotate_intrachromosomal_sv.R

```diff
@@ -0,0 +1,124 @@
+#' @title Annotate intrachromosomal SV.
+#'
+#' @description Annotate the intrachromosomal events in a data frame of SV breakpoints.
+#'
+#' @details Specify a data frame with SVs (preferably the output from get_manta_sv() to the `sv_data` parameter and get back the same data frame with SV annotations.
+#'
+#' @param sv_data A data frame of SVs. This should be the output of get_manta_sv.
+#' The required columns are columns for both chromosomes, coordinates and strand plus SOMATIC_SCORE and tumour_sample_id.
+#' @param return_as Stated format for returned output, default is "bedpe". Other accepted output formats are "bed" and "bedpe_entrez" (to keep entrez_ids for compatibility with portal.R and cBioPortal).
+#' @param projection Reference genome build parameter. Can be one of grch37 or hg38. The default is grch37.
+#' @param types A vector of sv types to be annotated. The default is set to "DEL", "DUP", "INS", and "INV".
+#' @param max_size An integer specifying the maximum distance between start of first breakpoint and end of second breakpoint. The default is 100000 (100000 bp).
+#' @param keep_genes Optional argument allowing ot only return svs annotated to the list of genes provided in this argument. The default is "all" (return all svs annotated to all genes).
+#' @param verbose Optionally control whether to print any status info to console. Useful for debugging. Default is FALSE (no messages).
+#'
+#'
+#' @return A data frame with annotated SVs (gene symbol and entrez ID).
+#'
+#' @rawNamespace import(data.table, except = c("last", "first", "between", "transpose"))
+#' @import tidyr dplyr stringr
+#' @export
+#'
+#' @examples
+#' sv_df = get_manta_sv(verbose = FALSE)
+#' annotated = annotate_intrachromosomal_sv(sv_data = sv_df)
+#'
+annotate_intrachromosomal_sv = function(
+    sv_data,
+    return_as = "bedpe",
+    projection = "grch37",
```

Comment:
There seems to be no way for the function to check the projection of the incoming `sv_df` compared to what is specified with the `projection` parameter. For example, what if the user gets mantas SVs with projection set to hg38, and then runs this function with `projection = "grch37"`, would this cause problems? I can see that the `projection` is only used for the internal function call (`get_affected_genes`) and nothing else.

Reply:
Right, but we also do not do that for other functions as well - for example, annotate_sv, annotate_hotspots etc. Part of the reason is that there is no accurate or straightforward way to check for the incoming projection. We can in theory check for coordinates of specific genes but that would assume those genes are present in the input data which may not always be the case.